src/pages/resultsView/coExpression/CoExpressionViz.tsx

```diff
@@ -160,31 +162,39 @@ export default class CoExpressionViz extends React.Component<ICoExpressionVizPro
 
     private getPlotDataPromises(yAxisCoExpression?:CoExpressionWithQ) {
         const ret:{
-            molecularX:MobxPromise<NumericGeneMolecularData[]>,
-            molecularY:MobxPromise<NumericGeneMolecularData[]>|undefined,
+            molecularX:MobxPromise<NumericGeneMolecularData[]|GenesetMolecularData[]>,
+            molecularY:MobxPromise<NumericGeneMolecularData[]|GenesetMolecularData[]>|undefined,
             mutationX:MobxPromise<Mutation[]>|undefined,
             mutationY:MobxPromise<Mutation[]>|undefined
         } = {
-            molecularX: this.props.numericGeneMolecularDataCache.get({
-                entrezGeneId: this.props.gene.entrezGeneId,
-                molecularProfileId: this.props.molecularProfile.molecularProfileId
+            molecularX: this.props.profileX.molecularAlterationType === 'GENESET_SCORE' ?
+                this.props.numericGenesetMolecularDataCache.get({
+                genesetId: this.props.geneticEntity.geneticEntityId.toString(),
+                molecularProfileId: this.props.profileX.molecularProfileId
+            }) : this.props.numericGeneMolecularDataCache.get({
+                entrezGeneId: Number(this.props.geneticEntity.geneticEntityId),
+                molecularProfileId: this.props.profileX.molecularProfileId
             }),
             mutationX:undefined,
             molecularY:undefined,
             mutationY:undefined
         };
 
         if (yAxisCoExpression) {
-            ret.molecularY = this.props.numericGeneMolecularDataCache.get({
-                entrezGeneId: yAxisCoExpression.entrezGeneId,
-                molecularProfileId: this.props.molecularProfile.molecularProfileId
+            ret.molecularY = this.props.profileY.molecularAlterationType === 'GENESET_SCORE' ? 
+            this.props.numericGenesetMolecularDataCache.get({
+                genesetId: yAxisCoExpression.geneticEntityId,
+                molecularProfileId: this.props.profileY.molecularProfileId
+            }) : this.props.numericGeneMolecularDataCache.get({
+                    entrezGeneId: parseFloat(yAxisCoExpression.geneticEntityId),
+                    molecularProfileId: this.props.profileY.molecularProfileId
             });
         }
 
         if (this.props.mutationCache) {
-            ret.mutationX = this.props.mutationCache.get({entrezGeneId: this.props.gene.entrezGeneId});
+            ret.mutationX = this.props.mutationCache.get({entrezGeneId: Number(this.props.geneticEntity.geneticEntityId)});
             if (yAxisCoExpression) {
-                ret.mutationY = this.props.mutationCache.get({entrezGeneId: yAxisCoExpression.entrezGeneId});
+                ret.mutationY = this.props.mutationCache.get({entrezGeneId: parseFloat(yAxisCoExpression.geneticEntityId)});
```

Comment:
entrezGeneId is an integer, so it should be `parseInt(..., 10)` - the 10 is important